src/renderer/atlas/BackendD3D.cpp

```diff
@@ -1963,6 +2010,10 @@ size_t BackendD3D::_drawCursorForegroundSlowPath(const CursorRect& c, size_t off
         return 0;
     }
 
+    // Line drawing primitives requires texcoord to remain the same. Using this
+    // flag, we'll also avoid the need for branching in the calculations below.

```

Comment:
This relies on the compiler understanding that the multiplication can be turned into a masking depending on a boolean. MSVC unfortunately is one of the compilers that doesn't get this.
Primitive ternary branches on the other hand can be optimized very well by all major compilers and pretty much always results in conditional moves: https://godbolt.org/z/e5W1x8d7b

The downside of conditional moves is that it results in data dependence and so the instructions can't be run out of order. But I don't think this matters here. By the time we do the conditional move, the value of `isStyledLineDrawing` will already be fully realized for sure.

Reply:
I think this can be resolved, since this code is removed now that we send the rendition scale in a different way.